### PR TITLE
fix ClassTypeResolverTest on windows

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/rectorphp/rector/issues/6468

### CODE REVIEW

File: packages-tests/NodeTypeResolver/PerNodeTypeResolver/ClassAndInterfaceTypeResolver/ClassTypeResolverTest.php [hunk 1]
```diff
@@ -46,11 +46,19 @@ public function dataProvider(): Iterator
...
-        yield [
-            __DIR__ . '/Source/AnonymousClass.php',
-            0,
-            new ObjectType('AnonymousClassf58ab370f3875d601f309c1728c0e151'),
-        ];
+        $resolvedType = $this->nodeTypeResolver->resolve($variableNodes[$nodePosition]);
+        $this->assertInstanceOf(TypeWithClassName::class, $resolvedType);
+
+        /** @var TypeWithClassName $resolvedType */
+        $this->assertStringStartsWith('AnonymousClass', $resolvedType->getClassName());
```
Comment (staabm) [inline packages-tests/NodeTypeResolver/PerNodeTypeResolver/ClassAndInterfaceTypeResolver/ClassTypeResolverTest.php:63]:
```suggestion
        // anonymous classes contain a hash, which is different between platforms
        $this->assertStringStartsWith('AnonymousClass', $resolvedType->getClassName());
```

**Comment by staabm** [comment]
ready to land

**Comment by staabm** [comment]
@TomasVotruba good to go

**Comment by TomasVotruba** [comment]
Thanks!

**Pull request merged**